image-loader/app/controllers/ImageLoaderController.scala

```diff
@@ -79,39 +79,51 @@ class ImageLoaderController(auth: Authentication,
     val parsedBody = DigestBodyParser.create(tempFile)
 
     AuthenticatedAndAuthorised.async(parsedBody) { req =>
-      val uploadTimeToRecord = DateTimeUtils.fromValueOrNow(uploadTime)
-      val uploadedByToRecord = uploadedBy.getOrElse(Authentication.getIdentity(req.user))
+      val onBehalfOfFn: OnBehalfOfPrincipal = auth.getOnBehalfOfPrincipal(req.user)
+      val isSoftDeleted = gridClient.getImageSoftDeletedStatus(req.body.digest, onBehalfOfFn)
+      isSoftDeleted.flatMap{
```

Comment:
An alternative way here would be to do something like this:
`val result = for {
    softDeleted <- isSoftDeleted
    _ <- if(softDeleted) Future.failed(ImageAlreadyExistsException) else Future.successful(true)
    ....`
    and then add a case in the recover section to handle that exception.